src/builder/QuarkBuilder.sol

```diff
@@ -52,6 +52,145 @@ contract QuarkBuilder {
     /* ===== Helper Functions ===== */
 
     /* ===== Main Implementation ===== */
+    struct CometRepayIntent {
+        uint256 amount;
+        string assetSymbol;
+        uint256 blockTimestamp;
+        uint256 chainId;
+        uint256[] collateralAmounts;
+        string[] collateralAssetSymbols;
+        address comet;
+        address repayer;
+    }
+
+    function cometRepay(
+        CometRepayIntent memory repayIntent,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) external pure returns (BuilderResult memory /* builderResult */ ) {
+        if (repayIntent.collateralAmounts.length != repayIntent.collateralAssetSymbols.length) {
+            revert InvalidInput();
+        }
+
+        // XXX confirm that the user is not withdrawing beyond their limits
+
+        assertFundsAvailable(
+            repayIntent.chainId, repayIntent.assetSymbol, repayIntent.amount, chainAccountsList, payment
+        );
+
+        uint256 actionIndex = 0;
+        Actions.Action[] memory actions = new Actions.Action[](chainAccountsList.length);
+        IQuarkWallet.QuarkOperation[] memory quarkOperations =
+            new IQuarkWallet.QuarkOperation[](chainAccountsList.length);
+
+        bool useQuotecall = false; // TODO: calculate an actual value for useQuoteCall
```

Comment:
Trying to think what would be a use case for quotecall in repay. 
I guess is when user has 10USDC,  and comet has -10USDC, 1ETH. 
So user trigger repayMax?